src/pages/stake/table.tsx

```diff
@@ -69,20 +87,61 @@ const TokensStakedCell: FunctionComponent<CellRendererProps<any, ParsedValidator
 	return <TokensCellWrapper className={value ? 'tokens' : 'no_tokens'}>{value || 'no tokens'}</TokensCellWrapper>;
 };
 
-const ActionCell: FunctionComponent<CellRendererProps<boolean, ParsedValidator>> = ({ data, value }) =>
-	value ? (
-		<ActionCellWrapper>
-			<ReDelegateButton valAddress={data.operatorAddress} />
-			<Divider />
-			<UnDelegateButton valAddress={data.operatorAddress} />
-			<Divider />
-			<DelegateButton valAddress={data.operatorAddress} />
-		</ActionCellWrapper>
-	) : (
+const ActionCell: FunctionComponent<CellRendererProps<boolean, ParsedValidator>> = ({ data, value }) => {
+	const { lang } = useAppSelector(selector);
+	const [handleOpen, showMessage] = useActions([
+		delegateDialogActions.showDelegateDialog,
+		snackbarActions.showSnackbar,
+	]);
+
+	const address = useAddress();
+
+	const handleClick = (currentlySelected: string) => {
+		if (!address) {
+			showMessage(variables[lang]['connect_account']);
+			return;
+		}
+
+		const actionPayload: ShowDelegateDialog = { name: 'Redelegate', validatorAddress: data.operatorAddress };
+
+		if (currentlySelected === 'Undelegate') {
```

Comment:
Can you just set `actionPayload.name` to `currentlySelected`?

Reply:
ah right, now that I removed the stake option. Updated this